apps/bbsync/query.py

```diff
@@ -389,3 +389,84 @@ def generate_comment(self):
                 "body": pending_comments.first().text,
                 "is_private": False,
             }
+
+    def generate_fixed_in(self):
+        """
+        Generates bugzilla query value for the bugzilla field fixed_in based
+        on the hierarchy of Flaw-associated models Package and PackageVer.
+
+        Except for flaws that have been stored in the database for a long time
+        and haven't been refreshed by bzimport since introducing this function
+        (and the associated save_packageversions in convertors.py). For such,
+        the bugzilla query for the flag is silently not generated.
+        """
+
+        if "fixed_in" not in self.flaw.meta_attr:
+            # TODO: Consider manually bzimporting leftover Flaws without
+            #       meta_attr["fixed_in"] before prod goes writable. It might be
+            #       a manageable number by then.
+            # TODO: [File an issue to track this, add the OSIDB-<number> ID here.]
+            #
+            # Do not send fixed_in to bugzilla. If this bbsync runs because of an API request
+            # to edit fixed_in, the change will be silently discarded.
+            #
+            # * If this branch executes, this flaw has not been updated in OSIDB database since
+            #   adding support for fixed_in. That's a long time.
+            # * OSIDB is planned to run for months with meta_attr["fixed_in"] support before the
+            #   first usage of the API for modification of fixed_in
+            #   (/flaw/<flaw_id>/package_versions). Therefore on most Flaws where users might need
+            #   modifying fixed_in, meta_attr["fixed_in"] will have already been bzimported by the
+            #   time they start doing so.
+            # * We can't just blanket raise an exception when meta_attr["fixed_in"] is missing
+            #   because that would block all bbsync operations for all not-yet-bzimported Flaws.
+            # * Detecting API-performed changes without flaw.meta_attr["fixed_in"] would be
+            #   nontrivial, so intelligently deciding when to raise such an exception would also be
+            #   nontrivial.
+            # * The most proper way would be to bzimport the Flaw here and populate meta_attr.
+            #   However, that is potentially complicated and breaks decomposition for a probably
+            #   miniscule benefit.
+            # * Any Flaw bbsync triggers a follow-up bzimport that will load meta_attr["fixed_in"].
+            #   We can expect that if an old Flaw gets updated, fixed_in will not be the first
+            #   edit operation.
+            #
+            return
```

Comment:
I think this situation cannot happen. You added `meta_attr["fixed_in"] = self.flaw_bug["fixed_in"]` in BZImport, and even if `fixed_in` will be empty in BZ, `meta_attr` in OSIDB will contain `'fixed_in': ''`.

Reply:
Why can't it happen? The comment is detailed and explains what can happen, so I'd expect you to provide similarly detailed reasoning why it's wrong.

`meta_attr["fixed_in"] = self.flaw_bug["fixed_in"]` in bzimport saves the verbatim bugzilla `fixed_in` field for _newly bzimported flaws_. It doesn't add `meta_attr["fixed_in"]` to existing flaws in the database.

> and even if fixed_in will be empty in BZ, meta_attr in OSIDB will contain 'fixed_in': ''.

- Empty string value for an existing dictionary key is substantially different than a nonexistent dictionary key.
- "Detecting API-performed changes without flaw.meta_attr["fixed_in"] would be nontrivial(...)"
- We can't just create a fake empty string out of nothing and send that to bugzilla if we don't have `meta_attr["fixed_in"]`. That would destroy data in bugzilla.